src/opm/output/eclipse/Summary.cpp

```diff
@@ -1257,16 +1236,17 @@ void eval_udq(const Opm::Schedule& schedule, std::size_t sim_step, Opm::SummaryS
     }
 }
 
-void updateValue(const Opm::SummaryConfigNode& node, const double value, Opm::SummaryState& st)
-{
-    if (node.category() == Opm::SummaryConfigNode::Category::Well)
-        st.update_well_var(node.namedEntity(), node.keyword(), value);
+void updateValue(const Opm::EclIO::SummaryNode& node, const double value, Opm::SummaryState& st) {
+    switch (node.category) {
+    case Opm::EclIO::SummaryNode::Category::Well:
+        return st.update_well_var(node.wgname, node.keyword, value);
 
-    else if (node.category() == Opm::SummaryConfigNode::Category::Group)
-        st.update_group_var(node.namedEntity(), node.keyword(), value);
+    case Opm::EclIO::SummaryNode::Category::Group:
+        return st.update_group_var(node.wgname, node.keyword, value);
 
-    else
-        st.update(node.uniqueNodeKey(), value);
+    default:
+        return st.update(node.unique_key(), value);
```

Comment:
`SummaryState::update()` does not return a value.  Pretending that it does will only lead to confusion.  Why do you need to use a switch here anyway.  The original was more readable.

Reply:
Reverted from `switch`.